MapboxCoreNavigation/NativeRouteController.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+import MapboxNavigationNative
+import MapboxDirections
+
+
+public class NativeRouteController: RouteController {
+    
+    let navigator = MBNavigator()
+    
+    override public var route: Route {
+        get {
+            return routeProgress.route
+        }
+        set {
+            routeProgress = RouteProgress(route: newValue)
+            updateNavigator()
+        }
+    }
+    
+    func updateNavigator() {
+        guard let json = route.json else {
+            assertionFailure("Missing json route")
```

Comment:
This is no better than an `assert(_:)` followed by a force-unwrap without a guard.